### DISCUSSION TITLE
How to use MacOS Appkit with AvaloniaUI ??

### DISCUSSION THREAD

**Discussion by tictac1234** [AvaloniaUI/Avalonia#12946] [General]
Hello.

(I'm Japanese, sorry if my English is not correct...)

First of all I want to thank the developers of AvaloniaUI !
This framework is fantastic!
Recently i found out about this and am very impressed.
I didn't know such a thing existed! (ChatGPT told me about it).

So I would like to use AvaloniaUI to create a desktop app on macOS.

Anyway, so far I was able to
install -> compile -> run AvaloniaApp
is OK.

But I think I need to use Xamarin.Mac to use NSOpenPanel, NSSavePanel.
So I wanted to use Appkit and have made the following modifications. (.csproj)

```
<TargetFramework>net6.0</TargetFramework>
↓
<TargetFramework>net6.0-macos</TargetFramework>
``` 

But I get an error and cannot compile....
(I have referred to other articles, but I don't know...)

The error is this ↓

```
/Sdk/targets/Microsoft.NET.Sdk.targets(3,3): Error MSB4019: The imported project /usr/local/share/dotnet/sdk/7.0.308/Sdks/Microsoft.NET.Sdk/12.3.2372/targets/Xamarin.Shared.Sdk.MultiTarget.targets" was not found. Confirm that the expression in the Import declaration ";. /12.3.2372/targets/Xamarin.Shared.Sdk.MultiTarget.targets" is correct, and that the file exists on disk. (MSB4019) (AvaloniaApp_Mac)
``` 

Did I do something wrong?
On the Windows version, it was easy to call the file selection dialog from AvaloniaApp,
I'm struggling with it on MacOS...

This is not a problem with the AvaloniaUI, though, this is clearly a problem with my lack of knowledge,
I would be very happy if you could tell me.
Or is there a sample or something that this can be implemented ?

Thank you!

**Comment by maxkatz6** [comment]
> The error is this ↓

Probably because you have WinExe as a output type.
For net6 I did this example - https://github.com/maxkatz6/avalonia-net6-macos-sample, but in net7 it should be a bit easier, some steps can be skipped.

**Comment by tictac1234** [comment]
Thank you for response 🙇🏻‍♂️
I read the description in your link.

The reason I couldn't use AppKit was that I didn't download MonoMac nuget package...

I downloaded this, and next I added "NSApplication.Init()" in Main method".
It was able to use NSAlert, NSOpenPanel, and more.

I will read it when I need it as I don't have an official release on MacOS yet.

Thank you very much your kindness !

<img width="664" alt="スクリーンショット 2023-09-20 11 45 23" src="https://github.com/AvaloniaUI/Avalonia/assets/84780155/5e915d57-0b25-4704-b57d-e0b816319c85">
